### PR TITLE
[receiver/hostmetrics] Add `process.paging.faults` metric

### PR DESCRIPTION
**Pull request by andrzej-stencel**
**Description:**

This pull request adds a new metric named `process.paging.faults` to the metrics scraped by the `process` scraper of the `hostmetrics` receiver.

Adding this metric to the semantic conventions has been proposed here:
- https://github.com/open-telemetry/opentelemetry-specification/issues/2809.

There's a related system metric [system.paging.faults](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.61.0/receiver/hostmetricsreceiver/internal/scraper/pagingscraper/metadata.yaml#L65) available in the `paging` scraper. 

**Link to tracking Issue:** 

- https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/14084

### CODE REVIEW

File: receiver/hostmetricsreceiver/internal/scraper/processscraper/documentation.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ These are the metrics available for this scraper.
 | **process.disk.io.write** | Disk bytes written. | By | Sum(Int) | <ul> </ul> |
 | **process.memory.physical_usage** | The amount of physical memory in use. | By | Sum(Int) | <ul> </ul> |
 | **process.memory.virtual_usage** | Virtual memory size. | By | Sum(Int) | <ul> </ul> |
+| **process.paging.faults** | Number of page faults the process has made. This metric is only available on Linux. | {faults} | Sum(Int) | <ul> <li>type</li> </ul> |
```
Comment (dmitryax) [inline receiver/hostmetricsreceiver/internal/scraper/processscraper/documentation.md:17]: Do you think this metric is important enough to be added as enabled by default? Maybe better to set it as optional? That would be a non-disruptive change

**Comment by dmitryax** [comment]
Please rebase

File: receiver/hostmetricsreceiver/internal/scraper/processscraper/documentation.md (hunk 1 above)
Comment (astencel-sumo) [reply receiver/hostmetricsreceiver/internal/scraper/processscraper/documentation.md:17]: No problem at all making it optional, I'll make this change.

**Comment by dmitryax** [comment]
@astencel-sumo PTAL at the failing tests

**Review by dmitryax — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**